File: Models/RouteModel.js
const mongoose = require("mongoose");

const routeSchema = new mongoose.Schema({
  name: {
    type: String,

  },
  routeNumber: {
    type: String,
    trim: true,
    unique: true,
    required:[true,'route number not set up']
  },
  Stops:[{
    type:mongoose.Schema.Types.ObjectId,
    ref:'stations'
}],
  startPoint:{
  type:String,
  required:[true,'start point not set up']

 },
 endPoint:{
  type:String,
  require:[true,'end point not set up']
  
 },
  isClosed: {
    type: Boolean,
    default:false
  },
  createdAt:{
    type:Date,
    default:Date.now


  }
});




const RouteModel = mongoose.model("routes", routeSchema);

module.exports = {
  
  RouteModel,

};

File: lib/Models/RouteModel.dart
class RouteModel {
  List<String>? stops;
  bool? isClosed;
  String? routeNumber;
  String? startPoint;
  String? endPoint;

  RouteModel(
      {this.stops,
        this.isClosed,
        this.routeNumber,
        this.startPoint,
        this.endPoint});

  RouteModel.fromJson(Map<String, dynamic> json) {
    if (json['Stops'] != null) {
      stops = <String>[];
      json['Stops'].forEach((v) {

        if(v!=null){
          stops!.add(v.toString());
        }

      });


    }
    isClosed = json['isClosed'];
    routeNumber = json['routeNumber'];
    startPoint = json['startPoint'];
    endPoint = json['endPoint'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();

    data['isClosed'] = this.isClosed;
    data['routeNumber'] = this.routeNumber;
    data['startPoint'] = this.startPoint;
    data['endPoint'] = this.endPoint;
    return data;
  }
}